zebra-network/src/peer_set/initialize.rs

```diff
@@ -135,16 +135,19 @@ where
 
     let listen_guard = tokio::spawn(listen(config.listen_addr, listener, peerset_tx.clone()));
 
+    // 2. Initial peers, specified in the config.
     let initial_peers_fut = {
-        let initial_peers = config.initial_peers();
+        let config = config.clone();
         let connector = connector.clone();
-        let tx = peerset_tx.clone();
-
-        // Connect the tx end to the 3 peer sources:
-        add_initial_peers(initial_peers, connector, tx)
+        let peerset_tx = peerset_tx.clone();
+        async move {
+            let initial_peers = config.initial_peers().await;
+            // Connect the tx end to the 3 peer sources:
+            add_initial_peers(initial_peers, connector, peerset_tx).await
+        }
+        .boxed()
```

Comment:
These clones are all needed, but if we put them in a block `{ ... }`, then we don't need to use special names for them.

The names are bound to the clones inside the block, but they are bound to the originals once the block ends.